pytorch_lightning/plugins/training_type/ipu.py

```diff
@@ -0,0 +1,325 @@
+        self.model = model
+
+        # Separate models are instantiated for different stages, but they share the same weights on host.
+        # When validation/test models are run, weights are synced first.
+
+        if self.lightning_module.trainer.training:
+            # Create model for training which will run training.
+            optimizer = self.lightning_module.trainer.optimizers[0]
+            model = poptorch.trainingModel(model=model, options=self._create_opts(training=True), optimizer=optimizer)
+            self.poptorch_models['train'] = model
+        for x in ('val', 'test', 'predict'):
+            model = poptorch.inferenceModel(
+                model=model,
+                options=self._create_opts(training=False),
+            )
+            self.poptorch_models[x] = model
+
+    @property
+    def replication_factor(self):
+        return len(self.parallel_devices)
+
+    def _create_opts(self, training: bool):
+        opts = poptorch.Options()
+        opts.deviceIterations(self.device_iterations)
+        opts.replicationFactor(self.replication_factor)
+        gradient_accumulation = self.lightning_module.trainer.accumulate_grad_batches if training else 1
+        opts.Training.gradientAccumulation(gradient_accumulation)
+        opts.autoRoundNumIPUs(self.autoround_num_ipus)
+
+        # todo (sean): unsure if this is necessary but to be safe.
+        if os.environ.get("PL_GLOBAL_SEED"):
+            opts.randomSeed(int(os.environ["PL_GLOBAL_SEED"]))
+        return opts
+
+    def on_reset_train_dataloader(self, dataloader: Union[Iterable, DataLoader]) -> Union[Iterable, DataLoader]:
+        return self.process_dataloader(dataloader)
+
+    def on_reset_val_dataloader(self, dataloader: Union[Iterable, DataLoader]) -> Union[Iterable, DataLoader]:
+        return self.process_dataloader(dataloader)
+
+    def on_reset_test_dataloader(self, dataloader: Union[Iterable, DataLoader]) -> Union[Iterable, DataLoader]:
+        return self.process_dataloader(dataloader)
+
+    def on_reset_predict_dataloader(self, dataloader: Union[Iterable, DataLoader]) -> Union[Iterable, DataLoader]:
+        return self.process_dataloader(dataloader)
+
+    def process_dataloader(self, dataloader: Union[Iterable, DataLoader]) -> Union[Iterable, DataLoader]:
+        if isinstance(dataloader, CombinedLoader):
+            dataloader.loaders = apply_to_collection(
+                dataloader.loaders,
+                DataLoader,
+                self.process_dataloader,
+            )
+            return dataloader
+        elif isinstance(dataloader, list):
+            dataloader = apply_to_collection(dataloader, DataLoader, self.process_dataloader)
+            return dataloader
+        if not isinstance(dataloader, poptorch.DataLoader):
+            dataloader = self._convert_to_poptorch_loader(
+                dataloader=dataloader, opts=self._create_opts(training=self.lightning_module.training)
```

Comment:
Might want to use the running_stage instead. It is more stable. User could decide to keep their model in train mode during validation.